File: P1_E3/VehicleBuilder.java
package P1_E3;

public abstract class VehicleBuilder {
    protected Vehicle vehicle;

    public Vehicle getVehicle() {
        return vehicle;
    }

    public void createNewVehicle() {
        vehicle = new Vehicle();
    }

    public abstract void buildEngine(String engine);
    public abstract void buildWheels();
    public abstract void buildColor(boolean customized, String color);
    public abstract void buildAudioSystem(boolean audioSystem);
    public abstract void buildTransmission(boolean transmission);
}

File: P2/VehicleBuilder.dart
// ignore_for_file: file_names, non_constant_identifier_names

import 'Vehicle.dart';

abstract class VehicleBuilder{
  late Vehicle vehicle;
  late double baseCost;

  VehicleBuilder({required this.baseCost});


  Vehicle getVehicle(){
    return vehicle;
  }

  void createNewVehicle(){
    vehicle = Vehicle();
  }

//Metodos abstractos
  void buildEngine(String engine);
  void buildWheels();
  void buildAudioSystem(bool audioSystem);
  void buildTransmission(bool transmission);
}